text_to_sql.py

```
@@ -0,0 +1,172 @@
+        return []
+
+    try:
+        with SessionLocal() as session:
+            logger.debug(f"Executing SQL: {sql}")
+            rows = session.execute(text(sql)).fetchall()
+            result = [dict(r._mapping) for r in rows]
+            logger.debug(f"SQL result: {result}")
+            return result
+    except Exception as e:
+        logger.error(f"SQL execution error: {e}")
+        raise ValueError(f"Failed to execute SQL: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)